cores/nRF5/WInterrupts.c

```diff
@@ -95,25 +96,63 @@ int attachInterrupt(uint32_t pin, voidFuncPtr callback, uint32_t mode)
       return 0;
   }
 
-  for (int ch = 0; ch < NUMBER_OF_GPIO_TE; ch++) {
-    if (channelMap[ch] == -1 || (uint32_t)channelMap[ch] == pin) {
-      channelMap[ch] = pin;
-      callbacksInt[ch] = callback;
-      callbackDeferred[ch] = deferred;
-
-      NRF_GPIOTE->CONFIG[ch] &= ~(GPIOTE_CONFIG_PORT_PIN_Msk | GPIOTE_CONFIG_POLARITY_Msk);
-      NRF_GPIOTE->CONFIG[ch] |= ((pin << GPIOTE_CONFIG_PSEL_Pos) & GPIOTE_CONFIG_PORT_PIN_Msk) |
-                              ((polarity << GPIOTE_CONFIG_POLARITY_Pos) & GPIOTE_CONFIG_POLARITY_Msk);
-
-      NRF_GPIOTE->CONFIG[ch] |= GPIOTE_CONFIG_MODE_Event;
-
-      NRF_GPIOTE->INTENSET = (1 << ch);
-
-      return (1 << ch);
+  // All information for the configuration is known, except the prior values
+  // of the config register.  Pre-compute the mask and new bits for later use.
+  //     CONFIG[n] = (CONFIG[n] & oldRegMask) | newRegBits;
+  //
+  // Three fields are configured here: PORT/PIN, POLARITY, MODE
+  const uint32_t oldRegMask = ~(GPIOTE_CONFIG_PORT_PIN_Msk | GPIOTE_CONFIG_POLARITY_Msk | GPIOTE_CONFIG_MODE_Msk);
```

Comment:
maybe move it near its usage. feel free to leave it as it is. I could do it later on.

Reply:
sure.